src/core/logical_layers/renderers/BivariateRenderer.tsx

```diff
@@ -102,17 +125,151 @@ export class BivariateRenderer extends LogicalLayerDefaultRenderer {
     if (layerData == null) return;
     this.mountBivariateLayer(map, layerData, legend);
 
+    const clickHandler = (ev: maplibregl.MapMouseEvent & maplibregl.EventData) => {
+      const features = ev.target
+        .queryRenderedFeatures(ev.point)
+        .filter((f) => f.source.includes(this._sourceId));
+      if (!features.length || !legend || !features[0].geometry) return true;
+      const geometry = features[0].geometry;
+
+      const fillColor: FillColor = features[0].layer.paint?.['fill-color'];
```

Comment:
if you're sure you need first feature why not create a variable for it and use it later?
`const feature = features[0]!`